gtda/homology/rescaling.py

```diff
@@ -163,12 +172,162 @@ def transform(self, X, y=None):
+    :math:`D`, then the corresponding entry in the transformed array is the
+    distance matrix :math:`D'_{ii+1} = factor \times D_{ii+1}`. If the entries
+    in `X` represent point clouds, their distance matrices are first computed,
+    and then rescaled according to the same formula.
+
+    Parameters
+    ----------
+    metric : string or callable, optional, default: ``'euclidean'``
+        If set to ``'precomputed'``, each entry in `X` along axis 0 is
+        interpreted to be a distance matrix. Otherwise, entries are
+        interpreted as feature arrays, and `metric` determines a rule with
+        which to calculate distances between pairs of instances (i.e. rows)
+        in these arrays.
+        If `metric` is a string, it must be one of the options allowed by
+        :func:`scipy.spatial.distance.pdist` for its metric parameter, or a
+        metric listed in :obj:`sklearn.pairwise.PAIRWISE_DISTANCE_FUNCTIONS`,
+        including "euclidean", "manhattan" or "cosine".
+        If `metric` is a callable function, it is called on each pair of
+        instances and the resulting value recorded. The callable should take
+        two arrays from the entry in `X` as input, and return a value
+        indicating the distance between them.
+
+    metric_params : dict, optional, default: ``None``
+        Additional keyword arguments for the metric function.
+
+    factor : float, optional, default: ``0.``
+        Factor by which to multiply the distance between consecutive
+        points.
+
+    n_jobs : int or None, optional, default: ``None``
+        The number of jobs to use for the computation. ``None`` means 1
+        unless in a :obj:`joblib.parallel_backend` context. ``-1`` means
+        using all processors.
+
+    Attributes
+    ----------
+    effective_metric_params_ : dict
+        Dictionary containing all information present in `metric_params`.
+        If `metric_params` is `None`, it is set to the empty dictionary.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from gtda.homology import ConsecutivetRescaling
+    >>> X = np.array([[[0, 0], [1, 2], [5, 6]]])
+    >>> cr = ConsecutiveRescaling()
+    >>> X_rescaled = cr.fit_transform(X)
+    >>> print(X_rescaled.shape)
+    (1, 3, 3)
+
+    See also
+    --------
+    VietorisRipsPersistence
+
+    """
+
+    _hyperparameters = {'factor': [numbers.Number, (0., np.inf)]}
+
+    def __init__(self, metric='euclidean', metric_params=None, factor=0.,
+                 n_jobs=None):
+        self.metric = metric
+        self.metric_params = metric_params
+        self.factor = factor
+        self.n_jobs = n_jobs
+
+    def _consecutive_homology_distance(self, X):
+        Xm = pairwise_distances(X, metric=self.metric, n_jobs=1,
+                                **self.effective_metric_params_)
+
+        Xm.ravel()[
+            1: max(0, Xm.shape[1] - 1) * Xm.shape[1]: Xm.shape[1] + 1] = 0
```

Comment:
Why do we set this distance  to 0? Shouldn't it be multiplied by `self.factor` ? ie, `*= self.factor` instead of ` = 0`.